libraries/consul_acl.rb

```diff
@@ -90,10 +90,35 @@ def configure_diplomat
       end
 
       def up_to_date?
-        old_acl = Diplomat::Acl.info(new_resource.to_acl['ID'], nil, :return)
-        return false if old_acl.nil? || old_acl.empty?
-        old_acl.first.select! { |k, _v| %w(ID Type Name Rules).include?(k) }
-        old_acl.first == new_resource.to_acl
+        retry_block(attempts: 2, catch: Diplomat::UnknownStatus, sleep: 0.5) do
+          old_acl = Diplomat::Acl.info(new_resource.to_acl['ID'], nil, :return)
+          return false if old_acl.nil? || old_acl.empty?
+          old_acl.first.select! { |k, _v| %w(ID Type Name Rules).include?(k) }
+          old_acl.first == new_resource.to_acl
+        end
+      end
+
+      def retry_block(opts = {}, &_block)
+        opts = {
+          sleep: 0, # Seconds to sleep between attempts
+        }.merge(opts)
+
+        opts[:catch] = [opts[:catch]].flatten
+        attempts = 1
+
+        begin
+          return yield attempts
+        rescue Diplomat::UnknownStatus
```

Comment:
Should it be `rescue *opts[:catch]` instead?

Reply:
No, this will be the only error we want to retry on

Reply:
Yeah, now it's fine